### ISSUE TITLE
Tailscale on linux doesn't recover from brief wifi outages

### ISSUE THREAD

**Issue by danderson** [tailscale/tailscale#1924]
Reported by a user: mosh session between two NixOS machines fails to recover after a brief (1-2s) wifi hiccup. Restarting tailscaled on the wifi-enabled machine leads to immediate mosh recovery.

Presumably we're failing to propagate network change events sufficiently.

Bug markers taken shortly after occurrences of loss of connectivity:

```
BUG-2bd92d2f82060c7bb68a796e8d30897357976d0bbb75ea6adb0d90c9118951e8-20210513075025Z-33c81dd5582dc734
BUG-2bd92d2f82060c7bb68a796e8d30897357976d0bbb75ea6adb0d90c9118951e8-20210513172751Z-45454d1603134967
BUG-2bd92d2f82060c7bb68a796e8d30897357976d0bbb75ea6adb0d90c9118951e8-20210513192642Z-960cf03cfbdbb413
```

**Comment by danderson** [comment]
Forgot to add: this is a regression from 1.6. 1.6 recovers correctly on this same setup.

**Comment by apenwarr** [comment]
We shouldn't need to propagate network change events, should we? The link should still recover automatically, even if it takes a few seconds.

This is a little different on darwin where sockets get invalidated on network change, but I've never heard of Linux doing that.

**Comment by crawshaw** [comment]
@danderson any thoughts on whether we should investigate this for 1.10?

**Comment by apenwarr** [comment]
I strongly suspect this is the "systemd wipes out policy routing settings when other unrelated network things change" bug, which I think is discussed in more detail elsewhere.

**Comment by DentonGentry** [comment]
https://github.com/tailscale/tailscale/issues/1699
https://github.com/tailscale/tailscale/issues/1591
are both likely.

**Comment by crawshaw** [comment]
@DentonGentry this bug is not listed in the "Must do 1.10.0" column of the Tailscale v1.10 project, but is listed in the v1.10.0 column of the "Serious bugs and regressions" project.

It appears to have been kicking around as a regression that was introduced 1.6->1.8. I don't remember the details but presumably it was not considered as affecting enough users to warrant getting done in a point release?

Either way, @danderson, do you want to decide if this is blocking 1.10.0 or not?

**Comment by DentonGentry** [comment]
I suspect this was fixed by either #1699 or #1591. Barring further updates, we'll expect to close this during the next bug scrub.

**Comment by DentonGentry** [comment]
Closing during bug scrub.
